package/setup.py

```diff
@@ -130,33 +129,118 @@ def get(self, option_name, default=None):
         except configparser.NoOptionError:
             return default
 
-class MDAExtension(Extension, object):
-    """Derived class to cleanly handle setup-time (numpy) dependencies.
+class MDADistribution(Distribution, object):
+    """Derived class to cleanly handle setup-time (cython/numpy) dependencies.
     """
-    # The only setup-time numpy dependency comes when setting up its
-    #  include dir.
-    # The actual numpy import and call can be delayed until after pip
-    #  has figured it must install numpy.
-    # This is accomplished by passing the get_numpy_include function
-    #  as one of the include_dirs. This derived Extension class takes
-    #  care of calling it when needed.
-    def __init__(self, *args, **kwargs):
-        self._mda_include_dirs = []
-        super(MDAExtension, self).__init__(*args, **kwargs)
-
-    @property
-    def include_dirs(self):
-        if not self._mda_include_dirs:
-            for item in self._mda_include_dir_args:
-                try:
-                    self._mda_include_dirs.append(item()) #The numpy callable
-                except TypeError:
-                    self._mda_include_dirs.append(item)
-        return self._mda_include_dirs
-
-    @include_dirs.setter
-    def include_dirs(self, val):
-        self._mda_include_dir_args = val
+    # Setup-time cython dependency comes when cythonizing sources (cython might
+    #  not be available) or configuring numpy's include dir (numpy might
+    #  not be available either).
+    # The actual cython/numpy imports and calls can be delayed until after
+    #  pip/setuptools have figured they must install them.
+    # This is accomplished by using a setuptools.dist.Distribution subclass
+    #  that lazily fills-in cython and numpy information by hooking into
+    #  the run_commands method.
+    # Our overrides also address the setuptools behavior of installing
+    #  setup-time dependencies locally. This involves essentially a rewrite
```

Comment:
Can you be more specific? I explain three concepts in this comment. Which should be clarified?

In any case, I already updated it a bit and it looks like this. Do comment if it's still not clear.
```
    # Setup-time dependences come when cythonizing sources (cython might
    #  not be available) or configuring numpy's include dir (numpy might
    #  not be available either).
    # The actual cython/numpy imports and calls can be delayed until after
    #  pip/setuptools have figured they must install them.
    # This is accomplished by using a setuptools.dist.Distribution subclass
    #  that lazily fills-in cython and numpy information by hooking into
    #  the run_commands method. run_commands is the last Distribution method
    #  before the actual installation takes place (the last method before  
    #  install that we can hook into by subclassing Distribution).
    # Our overrides also address the setuptools behavior of installing
    #  setup-time dependencies locally. This involves essentially a rewrite
    #  of fetch_build_egg, which is the method that takes care of installing
    #  dependencies on-the-fly.
    # Hopefully, since none of the overriden methods was pre-underscored,
    #  we'll stay clear of problems while this setuptools API lasts.
```